spyder/preferences/languageserver.py

```diff
@@ -608,30 +607,356 @@ def mouseDoubleClickEvent(self, event):
+            'pycodestyle/max_line_length', min_=10, max_=500, step=1,
+            tip=_("Default is 79"))
+
+        # Code style layout
+        code_style_g_layout = QGridLayout()
+        code_style_g_layout.addWidget(
+            self.code_style_filenames_match.label, 1, 0)
+        code_style_g_layout.addWidget(
+            self.code_style_filenames_match.textbox, 1, 1)
+        code_style_g_layout.addWidget(self.code_style_exclude.label, 2, 0)
+        code_style_g_layout.addWidget(self.code_style_exclude.textbox, 2, 1)
+        code_style_g_layout.addWidget(code_style_select.label, 3, 0)
+        code_style_g_layout.addWidget(code_style_select.textbox, 3, 1)
+        code_style_g_layout.addWidget(code_style_ignore.label, 4, 0)
+        code_style_g_layout.addWidget(code_style_ignore.textbox, 4, 1)
+        code_style_g_layout.addWidget(code_style_max_line_length.plabel, 5, 0)
+        code_style_g_layout.addWidget(
+            code_style_max_line_length.spinbox, 5, 1)
+
+        # Set Code style options enabled/disabled
+        code_style_g_widget = QWidget()
+        code_style_g_widget.setLayout(code_style_g_layout)
+        code_style_g_widget.setEnabled(self.get_option('pycodestyle'))
+        code_style_check.toggled.connect(code_style_g_widget.setEnabled)
+
+        # Code style layout
+        code_style_layout = QVBoxLayout()
+        code_style_layout.addWidget(code_style_label)
+        code_style_layout.addWidget(code_style_check)
+        code_style_layout.addWidget(code_style_g_widget)
+
+        code_style_widget = QWidget()
+        code_style_widget.setLayout(code_style_layout)
+
+        # --- Docstring tab ---
+        # Docstring style label
+        numpy_url = (
+            "<a href='https://numpydoc.readthedocs.io/en/"
+            "latest/format.html'>Numpy</a>")
+        pep257_url = (
+            "<a href='https://www.python.org/dev/peps/pep-0257/'>Pep 257</a>")
+        docstring_style_codes = _(
+            "<a href='http://www.pydocstyle.org/en/stable"
+            "/error_codes.html'>page</a>")
+        docstring_style_label = QLabel(
+            _("Here you can decide if you want to perform style analysis on "
+              "your docstrings according to the {} or {} conventions. You can "
+              "also decide if you want to show or ignore specific errors, "
+              "according to the codes found on this {}.").format(
+                  numpy_url, pep257_url, docstring_style_codes))
+        docstring_style_label.setOpenExternalLinks(True)
+        docstring_style_label.setWordWrap(True)
+
+        # Docstring style checkbox
+        docstring_style_check = self.create_checkbox(
+            _("Enable docstring style linting"),
+            'pydocstyle')
+
+        # Docstring style options
+        docstring_style_convention = self.create_combobox(
+            _("Choose the convention used to lint docstrings: "),
+            (("Numpy", 'numpy'),
+             ("Pep 257", 'pep257')),
```

Comment:
Assuming I understand what the various fields currently do correctly (and if not, that's a problem), to simplify the options and reduce duplication and confusion on this tab, I suggest including an option "Custom" in this list that makes the ``Show/ignore the following errors`` fields behave as exclusive overrides (i.e. *not* in addition to the specified convention), while otherwise they act in addition to such, and eliminate the extra set of duplicate fields.

Reply:
Also, ``PEP 257`` (compare ``Pep 257`` viz. ``PEP8``...) and ``Numpydoc``.

Reply:
> the fields Show the following errors and Ignore the following errors need to be disabled.

What I'm suggesting is instead of including two essentially duplicate sets of fields, only one of which is ever useful at a time, is to just have one set of fields (``Show in addition [...]``/``Ignore in addition``) and have Custom make them be in addition to nothing, i.e. an absolute whitelist/blacklist (as the lower set of fields, without the "in addition", does now). Otherwise, there isn't much point, IMO, in having Custom other than adding an extra step to do the same thing, as it adds rather than reduces complexity in our preferences for only a modest gain in clarity.

Reply:
Well, the problem is Pydocstyle and the PyLS make a difference between these options, so we need to configure them separately.

Ok, what about this:

1. Remove `Select/Ignore the following errors` options.
2. When Numpy or PEP 257 are selected, the labels of the options should be `Show/Ignore the following errors in addition to the specified convention`.
3. When `Custom` is selected, we change the text of those labels to `Show/ignore the following errors`.
4. We set the right options under the hood.

Reply:
:+1: Agreed, that's exactly what I'm suggesting.